drivers/slipdev/slipdev.c

```diff
@@ -220,6 +225,83 @@ static void _isr(netdev_t *netdev)
     }
 }
 
+#if IS_USED(MODULE_SLIPDEV_STDIO)
+static int _check_state(slipdev_t *dev)
+{
+    (void)dev;
+    return 0;
+}
+#else
+static void _poweron(slipdev_t *dev)
+{
+    if ((dev->state != SLIPDEV_STATE_STANDBY) ||
+        (dev->state != SLIPDEV_STATE_SLEEP)) {
+        return;
+    }
+
+    dev->state = 0;
+    uart_init(dev->config.uart, dev->config.baudrate, _slip_rx_cb, dev);
+}
+
+static void _poweroff(slipdev_t *dev, uint8_t state)
+{
+    uart_poweroff(dev->config.uart);
+    dev->state = state;
+}
+
+static int _check_state(slipdev_t *dev)
+{
+    /* discard data when interface is in SLEEP mode */
+    if (dev->state == SLIPDEV_STATE_SLEEP) {
+        return -ENETDOWN;
+    }
+
+    /* sending data wakes the interface from STANDBY */
+    if (dev->state == SLIPDEV_STATE_STANDBY) {
+        _poweron(dev);
+    }
+
+    return 0;
+}
```

Comment:
To make future changes to this function easier, I'd say do it like this:

```suggestion
static int _check_state(slipdev_t *dev)
{
    /* power states not supported when multiplexing stdio */
    if (IS_USED(MODULE_SLIPDEV_STDIO) {
        return 0;
    }
    /* discard data when interface is in SLEEP mode */
    if (dev->state == SLIPDEV_STATE_SLEEP) {
        return -ENETDOWN;
    }

    /* sending data wakes the interface from STANDBY */
    if (dev->state == SLIPDEV_STATE_STANDBY) {
        _poweron(dev);
    }

    return 0;
}
```

and move it out of the pre-processor guard.